packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -472,6 +470,9 @@ export class TreeCheckout implements ITreeCheckoutFork {
 			const { change, revision } = commit;
 			let withinEventContext = true;
 
+			// eslint-disable-next-line @typescript-eslint/no-this-alias
+			const outerThis = this;
```

Comment:
Still looking into overriding this pattern (1/2).

Reply:
I remember this coming up when I first started learning React. IIRC, you should be able to use arrow functions. Arrow functions don't have their own this context. Instead, they lexically inherit `this` from the surrounding code and you don't need to use .bind() to manually ensure that this remains consistent.